lib/meson.build

```diff
@@ -0,0 +1,29 @@
+libs = ['sdt_task']
+
+objs = []
+
+foreach lib : libs
+  bpf_o = gen_bpf_o.process(lib + '.bpf.c', extra_args: bpf_includes)
+  tgt = custom_target(lib,
+    output: lib + '.bpf.o',
+    input: lib + '.bpf.c',
+    command: [bpf_clang, bpf_base_cflags, bpf_includes, '-target', 'bpf',
+        libbpf_c_headers, '-c', '@INPUT@', '-o', '@OUTPUT@'],
+    depends: [libbpf],
+    build_by_default: true,
+    build_always_stale: true,
+  )
+
+  objs += [tgt]
+endforeach
+
+custom_target(scx_lib_name,
+  output: scx_lib_name + '.bpf.o',
```

Comment:
Would this work if there are two separate `util.bpf.c` files?

Reply:
Unfortunately no, the library compiles the files specified in a target list at the top of the lib/meson.build and Meson does not allow us including paths in that list. To do that we would have to add subdirectories in lib/.